src/MongoDB.Driver/Linq/Translators/PredicateTranslator.cs

```diff
@@ -1653,6 +1655,16 @@ private IFieldExpression GetFieldExpression(Expression expression)
             return fieldExpression;
         }
 
+        private void ValidatePipelineExpressionThrowIfNotValid(WhereExpression whereExpression)
+        {
+            var memberInfoCollection = ExpressionInfoCollector.Collect(whereExpression)?.ToList();
+            var unsupportedField = memberInfoCollection?.FirstOrDefault(c => !string.IsNullOrWhiteSpace(c.OutOfCurrentScopePrefix));
+            if (!string.IsNullOrWhiteSpace(unsupportedField?.OutOfCurrentScopePrefix))
```

Comment:
I think we already checked this... all we need to do in this if condition is check if unsupportedField != null.